apps/st2-packs/store.js

```diff
@@ -0,0 +1,244 @@
+          };
+        case 'error':
+          return {
+            ...state
+          };
+        default:
+          return {
+            ...state
+          };
+      }
+
+    case 'FETCH_PACK_INDEX':
+
+      switch(action.status) {
+        case 'success':
+          const packs = { ...state.packs };
+
+          _.forEach(action.payload, pack => {
+            packs[pack.ref] = {
+              ...state.packs[pack.ref],
+              ...pack
+            };
+          });
+
+          return {
+            ...state,
+            packs
+          };
+        case 'error':
+          return {
+            ...state
+          };
+        default:
+          return {
+            ...state
+          };
+      }
+
+      return {
+        ...state
+      };
+
+      case 'FETCH_PACK_CONFIG_SCHEMAS':
+
+        switch(action.status) {
+          case 'success':
+            const packs = { ...state.packs };
+
+            _.forEach(action.payload, pack => {
+              packs[pack.ref] = {
+                ...state.packs[pack.ref],
+                ...pack
+              };
+            });
+
+            return {
+              ...state,
+              packs
+            };
+          case 'error':
+            return {
+              ...state
+            };
+          default:
+            return {
+              ...state
+            };
+        }
+
+        return {
+          ...state
+        };
+
+      case 'FETCH_PACK_CONFIGS':
+
+        switch(action.status) {
+          case 'success':
+            const packs = { ...state.packs };
+
+            _.forEach(action.payload, pack => {
+              packs[pack.ref] = {
+                ...state.packs[pack.ref],
+                ...pack
+              };
+            });
+
+            return {
+              ...state,
+              packs
+            };
+          case 'error':
+            return {
+              ...state
+            };
+          default:
+            return {
+              ...state
+            };
+        }
+
+        return {
+          ...state
+        };
+
+    case 'SELECT_PACK':
+      const { ref } = action;
+
+      return {
+        ...state,
+        selected: ref || Object.keys(state.packs)[0]
+      };
+
+    case 'SET_FILTER':
+      const { filter } = action;
+
+      return {
+        ...state,
+        filter
+      };
+
+    default:
+      return state;
+  }
+};
+
+const promiseMiddleware = () => next => action => {
+  if (!action.promise) {
+    return next(action);
+  }
+
+  function actionFactory(status, data) {
+    const { promise, ...newAction } = action;
+    return {
+      ...newAction,
+      status,
+      ...data
+    };
+  }
+
+  next(actionFactory());
+  return action.promise.then(
+    payload => next(actionFactory('success', { payload })),
+    error => next(actionFactory('error', { error }))
+  );
+};
+
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const store = createStore(
```

Comment:
What's your plan long-term here as we use redux in more and more components? We will eventually need to think through how to centrally locate store and have scenes contain their own reducers and join them in the central store. Can we start to implement some of that in this PR? I think it would make sense to leverage redux for the TZ stuff we discussed this morning.

Thoughts?